api/envoy/config/filter/network/redis_proxy/v2/redis_proxy.proto

```diff
@@ -49,5 +55,59 @@ message RedisProxy {
   }
 
   // Network settings for the connection pool to the upstream cluster.
-  ConnPoolSettings settings = 3 [(validate.rules).message.required = true];
+  //
+  // .. attention::
+  //
+  //   This field is deprecated. Use a :ref:`catch-all
+  //   route<envoy_api_field_config.filter.network.redis_proxy.v2.RedisProxy.RouteConfig.catch_all>`
+  //   instead.
+  ConnPoolSettings settings = 3 [deprecated = true];
+
+  message PrefixMatch {
+    // If specified, the route is a prefix rule that must match the beginning of the keys. The proxy
+    // will always favor the longest matches.
+    string prefix = 1 [(validate.rules).string.min_bytes = 1];
+
+    // Indicates that prefix matching should be case insensitive.
+    bool case_insensitive = 2;
+  }
+
+  message ForwardAction {
+    // Name of cluster from cluster manager. See the :ref:`configuration section
+    // <arch_overview_redis_configuration>` of the architecture overview for recommendations on
+    // configuring the backing cluster.
+    string cluster = 1 [(validate.rules).string.min_bytes = 1];
+
+    // Network settings for the connection pool to the upstream cluster.
+    ConnPoolSettings settings = 2;
```

Comment:
On a second read through, it strikes me as a little odd that we have a per route/action connection pool. I think this is probably fine, but do we envision potentially having different routes go to the same cluster? Should we consider actually just having a single ConnPoolSettings that is used globally vs. per forward action? Or maybe it should be per cluster in some other map? Thoughts?

I'm also wondering if the code should be structured which maintains the connection pool map independently from the route table? Thoughts?